File: services/gateway/internal/dto/account.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type CreateAccountResponse struct {
	CompanyId uuid.UUID `json:"company_id"`
	Currency  string    `json:"currency"`
}

type LinkBankAccountRequest struct {
	AccountId         uuid.UUID `json:"account_id"`
	Name              string    `json:"name"`
	BIC               string    `json:"bic"`
	SettlementAccount string    `json:"settlement_account"`
	Currency          string    `json:"currency"`
}

type ReserveOperationRequest struct {
	AccountId     uuid.UUID `json:"account_id"`
	TransactionId uuid.UUID `json:"transaction_id"`
	Amount        int64     `json:"amount"`
}

type MakeBankOperationRequest struct {
	AccountId      uuid.UUID `json:"account_id"`
	BankAccountId  uuid.UUID `json:"bank_account_id"`
	Amount         int64     `json:"amount"`
	IdempotencyKey string    `json:"idempotency_key"`
}

type GenerateStatementRequest struct {
	PeriodFrom time.Time `json:"period_from"`
	PeriodTo   time.Time `json:"period_to"`
}

// RESPONSES
type AccountResponse struct {
	AccountId uuid.UUID `json:"account_id"`
	CompanyId uuid.UUID `json:"company_id"`
	Balance   int64     `json:"balance"`
	Currency  string    `json:"currency"`
	Status    string    `json:"status"`
	CreatedAt time.Time `json:"created_at"`
}

type BankAccountResponse struct {
	BankAccountId     uuid.UUID `json:"bank_account_id"`
	CompanyId         uuid.UUID `json:"company_id"`
	Name              string    `json:"name"`
	BIC               string    `json:"bic"`
	SettlementAccount string    `json:"settlement_account"`
	Currency          string    `json:"currency"`
	CreatedAt         time.Time `json:"created_at"`
}

type AccountOperationResponse struct {
	OperationId     uuid.UUID `json:"operation_id"`
	AccountId       uuid.UUID `json:"account_id"`
	TransactionId   uuid.UUID `json:"transaction_id"`
	OperationType   string    `json:"operation_type"`
	OperationStatus string    `json:"operation_status"`
	Amount          int64     `json:"amount"`
	CreatedAt       time.Time `json:"created_at"`
}

type BankOperationResponse struct {
	BankOperationId uuid.UUID `json:"bank_operation_id"`
	AccountId       uuid.UUID `json:"account_id"`
	BankAccountId   uuid.UUID `json:"bank_account_id"`
	InitiatorId     uuid.UUID `json:"initiator_id"`
	OperationType   string    `json:"operation_type"`
	OperationStatus string    `json:"operation_status"`
	Amount          int64     `json:"amount"`
	IdempotencyKey  string    `json:"idempotency_key"`
	ExternalId      string    `json:"external_id"`
	CreatedAt       time.Time `json:"created_at"`
}

type StatementEntry struct {
	Date         time.Time `json:"date"`
	EntryType    string    `json:"entry_type"`
	Amount       int64     `json:"amount"`
	BalanceAfter int64     `json:"balance_after"`
	Counterparty string    `json:"counterparty"`
}

type StatementResponse struct {
	StatementId    uuid.UUID        `json:"statement_id"`
	AccountId      uuid.UUID        `json:"account_id"`
	CompanyId      uuid.UUID        `json:"company_id"`
	InitiatorId    uuid.UUID        `json:"initiator_id"`
	PeriodFrom     time.Time        `json:"period_from"`
	PeriodTo       time.Time        `json:"period_to"`
	OpeningBalance int64            `json:"opening_balance"`
	ClosingBalance int64            `json:"closing_balance"`
	TotalDebit     int64            `json:"total_debit"`
	TotalCredit    int64            `json:"total_credit"`
	Currency       string           `json:"currency"`
	Entries        []StatementEntry `json:"entries"`
}

File: frontend/src/models/account.ts
export interface Account {
    account_id: string
    company_id: string
    balance: number
    currency: string
    status: string
    created_at: string
}

export interface BankAccount {
    bank_account_id: string
    company_id: string
    name: string
    bic: string
    settlement_account: string
    currency: string
    created_at: string
}

export interface BankOperation {
    bank_operation_id: string
    account_id: string
    bank_account_id: string
    operation_type: string
    operation_status: string
    amount: number
    idempotency_key: string
    external_id: string
    created_at: string
}

export interface CreateAccountRequest {
    currency: string
}

export interface LinkBankAccountRequest {
    account_id: string
    name: string
    bic: string
    settlement_account: string
    currency: string
}

export interface BankOperationRequest {
    account_id: string
    bank_account_id: string
    amount: number
    idempotency_key: string
}

export interface StatementEntry {
    date: string
    entry_type: string
    amount: number
    balance_after: number
    counterparty: string
}

export interface Statement {
    statement_id: string
    account_id: string
    company_id: string
    initiator_id: string
    period_from: string
    period_to: string
    opening_balance: number
    closing_balance: number
    total_debit: number
    total_credit: number
    currency: string
    entries: StatementEntry[]
    created_at: string
}

export interface GenerateStatementRequest {
    period_from: string
    period_to: string
}